modules/wechat_qrcode/src/wechat_qrcode.cpp

```diff
@@ -173,11 +183,11 @@ int WeChatQRCode::Impl::applyDetector(const Mat& img, vector<Mat>& points) {
     int img_w = img.cols;
     int img_h = img.rows;
 
+    const float targetArea = 400.f * 400.f;
     // hard code input size
-    int minInputSize = 400;
-    float resizeRatio = sqrt(img_w * img_h * 1.0 / (minInputSize * minInputSize));
-    int detect_width = img_w / resizeRatio;
-    int detect_height = img_h / resizeRatio;
+    const float tmpScaleFactor = scaleFactor == -1.f ? sqrt(targetArea / (img_w * img_h)) : scaleFactor;
```

Comment:
it'll do upscale for small images like qvga 320x240. I propose to use original image size for it.